mailbox/api/src/main/java/org/apache/james/mailbox/model/SimpleMailboxAnnotation.java

```diff
@@ -0,0 +1,328 @@
+    }
+
+    /**
+     * Creates a new instance of SimpleMailboxAnnotation containing no entries.
+     * 
+     */
+    public SimpleMailboxAnnotation() {
+        this.entries = Collections.emptyMap();
+    }
+
+    /**
+     * Creates a new instance of SimpleMailboxAnnotation from the given array of
+     * entries.
+     * 
+     * @param entries
+     */
+    public SimpleMailboxAnnotation(Map.Entry<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue>[] entries) {
+        if (entries != null) {
+            Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> m = 
+                    new HashMap<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue>(entries.length + entries.length / 2 + 1);
+            for (Entry<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> en : entries) {
+                m.put(en.getKey(), en.getValue());
+            }
+            this.entries = Collections.unmodifiableMap(m);
+        } else {
+            this.entries = Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Creates a new instance of SimpleMailboxAnnotation from the given {@link Map} of
+     * entries.
+     * 
+     * @param entries
+     */
+    public SimpleMailboxAnnotation(Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> entries) {
+        if (entries != null && entries.size() > 0) {
+            Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> m = 
+                    new HashMap<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue>(entries.size() + entries.size() / 2 + 1);
+            for (Entry<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> en : entries.entrySet()) {
+                m.put(en.getKey(), en.getValue());
+            }
+            this.entries = Collections.unmodifiableMap(m);
+        } else {
+            this.entries = Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Default implementation of {@link MailboxACLEntryKey}.
+     */
+    public static class SimpleMailboxAnnotationEntryKey implements MailboxAnnotationEntryKey {
+        private final String name;
+
+        /**
+         * Creates a new instance of SimpleMailboxAnnotationEntryKey from the given
+         * serialized {@link String}.
+         * 
+         * @param serialized
+         */
+        public SimpleMailboxAnnotationEntryKey(String serialized) {
+
+            if (serialized == null) {
+                throw new IllegalStateException("Cannot parse null to a " + getClass().getName());
+            }
+            serialized = serialized.trim();
+            if (serialized.length() == 0) {
+                throw new IllegalStateException("Cannot parse an empty string to a " + getClass().getName());
+            }
+            if (serialized.charAt(0) != SLASH_CHARACTER) {
+                throw new IllegalStateException("The annotation does not start with a slash " + getClass().getName());
+            }
+            if (serialized.contains("" + ASTERISK_CHARACTER)) {
```

Comment:
we never use string concatenation to do type conversion. If you want to go from a character to a String, call a method for that (such as toString)